src/main/webapp/app/app.component.ts

```diff
@@ -26,6 +26,10 @@ export class AppComponent implements OnInit {
     this.tasks$ = this.taskService.getAll();
   }
 
+  search(query): void {
+    this.tasks$ = this.taskService.getAll(query);
```

Comment:
would prefer to add a new function when searching or e.g. something like` this.tasks$ = query ? this.taskService.searchTasks(query) : this.taskService.getAll();`